src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketApiConnector.java

```diff
@@ -74,30 +77,48 @@ public BitbucketApi create(String owner, StandardUsernamePasswordCredentials cre
 
     @CheckForNull 
     public <T extends StandardCredentials> T lookupCredentials(@CheckForNull SCMSourceOwner context, @CheckForNull String id, Class<T> type) {
-        if (Util.fixEmpty(id) == null) {
-            return null;
-        } else {
-            if (id != null) {
-                return CredentialsMatchers.firstOrNull(
-                          CredentialsProvider.lookupCredentials(type, context, ACL.SYSTEM,
-                          bitbucketDomainRequirements()), 
-                          CredentialsMatchers.allOf(
-                              CredentialsMatchers.withId(id),
-                              CredentialsMatchers.anyOf(CredentialsMatchers.instanceOf(type))));
-            }
-            return null;
+        if (StringUtils.isNotBlank(id)) {
+            return CredentialsMatchers.firstOrNull(
+                      CredentialsProvider.lookupCredentials(
+                              type,
+                              context,
+                              context instanceof Queue.Task
+                                      ? Tasks.getDefaultAuthenticationOf((Queue.Task) context)
+                                      : ACL.SYSTEM,
+                              bitbucketDomainRequirements()
+                      ),
+                      CredentialsMatchers.allOf(
+                          CredentialsMatchers.withId(id),
+                          CredentialsMatchers.anyOf(CredentialsMatchers.instanceOf(type))
+                      )
+            );
         }
+        return null;
     }
 
-    public ListBoxModel fillCheckoutCredentials(StandardListBoxModel result, SCMSourceOwner context) {
-        result.withMatching(bitbucketCheckoutCredentialsMatcher(), CredentialsProvider.lookupCredentials(
-                StandardCredentials.class, context, ACL.SYSTEM, bitbucketDomainRequirements()));
+    public StandardListBoxModel fillCheckoutCredentials(StandardListBoxModel result, SCMSourceOwner context) {
+        result.includeMatchingAs(
+                context instanceof Queue.Task
+                ? Tasks.getDefaultAuthenticationOf((Queue.Task) context)
+                : ACL.SYSTEM,
+                context,
+                StandardCredentials.class,
+                bitbucketDomainRequirements(),
+                bitbucketCheckoutCredentialsMatcher()
+        );
         return result;
     }
 
-    public ListBoxModel fillCredentials(StandardListBoxModel result, SCMSourceOwner context) {
-        result.withMatching(bitbucketCredentialsMatcher(), CredentialsProvider.lookupCredentials(
-                StandardUsernameCredentials.class, context, ACL.SYSTEM, bitbucketDomainRequirements()));
+    public StandardListBoxModel fillCredentials(StandardListBoxModel result, SCMSourceOwner context) {
+        result.includeMatchingAs(
+                context instanceof Queue.Task
```

Comment:
All current implementations of `SCMSourceOwner` are a `Queue.Task` but there is not requirement that they be so